Rules/UseSingularNouns.cs

```diff
@@ -39,13 +46,21 @@ public class CmdletSingularNoun : IScriptRule {
         /// <param name="ast"></param>
         /// <param name="fileName"></param>
         /// <returns></returns>
-        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName) {
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
             if (ast == null) throw new ArgumentNullException(Strings.NullCommandInfoError);
 
             IEnumerable<Ast> funcAsts = ast.FindAll(item => item is FunctionDefinitionAst, true);
 
             char[] funcSeperator = { '-' };
             string[] funcNamePieces = new string[2];
+            
+#if !CORECLR
+            var usCultureInfo = CultureInfo.GetCultureInfo("en-us");
+            var pluralizationService = System.Data.Entity.Design.PluralizationServices.PluralizationService.CreateService(usCultureInfo);
+#else
+            var pluralizationService = new Pluralizer();
+#endif
```

Comment:
My main issue here is that we have two variables of the same name but different type. That works but I think is potentially too confusing/clever -- my preference is to avoid needing to juggle if-defs in my head if at all possible.

So given that my next question is whether both the `PluralizationService` and the `Pluralizer` are intended to have the same lifetime, what overhead is required to stand them up, and are they threadsafe?

Basically if they're threadsafe, I'd do this:

```csharp
#if CoreCLR
    private static readonly Pluralizer s_pluralizer = new Pluralizer();
#else
    private static readonly PluralizationService s_pluralizationService = PluralizationService.CreateService(CultureInfo.GetCultureInfo("en-US"));
#endif

private bool ShouldBeSingular(string noun)
{
#if CoreCLR
    return !s_pluralizer.IsSingular(noun) && s_pluralizer.IsPlural(noun);
#else
    return !s_pluralizationService.IsSingular(noun) && s_pluralizationService.IsPlural(noun);
#endif
}
```